### ISSUE TITLE
Bugfix Multigrid

### ISSUE THREAD

**Issue by TimoH447** [hpc2-2024/programming-project-awesome_team#16]
Check convergence

**Comment by TimoH447** [comment]
Possible Areas of mistake:

- Smoothing:
    Jacobi on its own converges, thus seems  correct
- Prolongation
- Restriction
- Gaussian elimination:
    Probably right, tested by doing multigrid with only one level for small N = > solution is exact in this case (no error)

**Comment by TimoH447** [comment]
Prolongation and Restriction had index error which are now corrected. But the multigrid is still not converges as fast as expected

**Comment by TimoH447** [comment]
Issue will be resolved with "debug" branch.

At the end what needed to be changed was:
- missing braces which led to running multigrid with grid size N which did not work for the number of levels
- changed the h=1/(N+1) from "being" on the right hand side to the matrix of the poisson problem
- main problem: bad convergence of jacobi -> fixed by using weighted jacobi

Due to chaotic debugging we added a lot bugs which we had to again "refix".
But now we get the expected fast convergence with usually <20 v-cycles.
